pages/moneygpt/index.js

```diff
@@ -0,0 +1,121 @@
+import { useTransactionsContext } from "@/utils/TransactionsContext/TransactionsContext";
+import Image from "next/image";
+import { useEffect, useState, useRef } from "react";
+import styled from "styled-components";
+import ReactMarkdown from "react-markdown";
+import { RotateCw } from "lucide-react";
+
+export default function Chat() {
+  const { data: transactions, isLoading } = useTransactionsContext();
+  const [response, setResponse] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [reloadTrigger, setReloadTrigger] = useState(0);
+
+  useEffect(() => {
+    const sendTransactions = async () => {
+      if (!transactions || transactions.length === 0) return;
+
+      setLoading(true);
+      setResponse("");
+
+      try {
+        const res = await fetch("/api/financial-analysis", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify(transactions),
+        });
+
+        const data = await res.json();
+        setResponse(data.response || "Keine Antwort erhalten.");
+      } catch (error) {
+        setResponse("Fehler: " + error.message);
+      }
+
+      setLoading(false);
+    };
+
+    sendTransactions();
+  }, [reloadTrigger]);
+
+  if (isLoading) {
+    return <p>Lade Daten</p>;
+  }
```

Comment:
I often read this in your project. But always I am asking myself if this is really necessary. Have you ever evaluated this? I am wondering because the TransactionProvider handles this case already internally and returns 'null' in that case. So I think it will always pass 'false' for the loading state. Do you know what I mean?